lib/pbench/test/unit/agent/test_validate_hostname.py

```diff
@@ -0,0 +1,15 @@
+"""Tests for validate_hostname
+"""
+
+from pbench.agent.utils import validate_hostname
+
+
+def test_validate_hostname():
+    assert validate_hostname("test") == 0
+    assert validate_hostname("test.example.com") == 0
+    assert validate_hostname("tes_t.example.com") == 1
```

Comment:
Funny; I had never actually realized (or at least didn't remember having known ;-) ) that "_" wasn't a valid hostname character... but I confirmed with the RFC. So, huh.

One comment on the tests, however: you should use some capital letters. Hostnames are case-insensitive, but we should be able to accept uppercase and it wouldn't be a bad thing to test that explicitly. Capital hex letters in the IPv6 wouldn't be bad, too.

Reply:
Yes, I didn't remember it either, but they did that right :-) I hate typing underscores... But they probably didn't do it to satisfy me: does the RFC contain a rationale?

Reply:
I could not find a rationale.  I think they started with what they already had for names in ARPAnet (which was a flat, global namespace), and made it hierarchical by adding dot-separated domains with similar naming conventions.